src/history/hs.c

```diff
@@ -348,6 +348,53 @@ __wt_hs_cursor_close(WT_SESSION_IMPL *session, WT_CURSOR **cursorp, uint32_t ses
     return (ret);
 }
 
+/*
+ * __hs_read_cursor_open --
+ *     Open a reading cursor to the history store. This function opens a new cursor through the user
+ *     session in order to inherit the visibility of the running transaction.
+ */
+static int
+__hs_read_cursor_open(WT_SESSION_IMPL *session, WT_CURSOR **cursorp, uint32_t *session_flags)
```

Comment:
@sulabhM is doing some work to remove the history store cursor pool so I'm trying not to step on his toes here. Once this code has stabilized a bit, it might be good to keep an open reading cursor on each session instead of opening a new one each time.